backend/src/routes/api/status/statusUtils.ts

```diff
@@ -18,32 +11,12 @@ export const status = async (
   const kubeContext = fastify.kube.currentContext;
   const { currentContext, namespace, currentUser, clusterID, clusterBranding } = fastify.kube;
   const customObjectsApi = fastify.kube.customObjectsApi;
-  const coreV1Api = fastify.kube.coreV1Api;
-  let isAdmin = false;
-
-  const userName = await getUserName(fastify, request, customObjectsApi);
 
-  try {
-    const groupConfig = await getGroupsConfig(coreV1Api, namespace);
-    const adminGroup = await getGroupsAdminConfig(coreV1Api, namespace, groupConfig);
-
-    if (adminGroup === SYSTEM_AUTHENTICATED || adminGroup === '') {
-      throw new Error(
-        'It is not allowed to set system:authenticated or an empty string as admin group.',
-      );
-    } else {
-      const adminUsers = await getGroup(customObjectsApi, adminGroup);
-      isAdmin = adminUsers?.includes(userName) ?? false;
-    }
-  } catch (e) {
-    fastify.log.error(e.toString());
-  }
+  const userName = await getUserName(request, customObjectsApi, currentUser);
+  const isAdmin = await isUserAdmin(fastify, userName, namespace);
```

Comment:
Hah, I think we did similar logic... might be an interesting conflict resolution.

[Current logic for admin](https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/routes/api/status/statusUtils.ts#L54)

Reply:
Yeah, it's been a journey 😅